### PR TITLE
Generated api

### PR DESCRIPTION
**Pull request by crast**
Provide the beginnings of a Generated API 
* some basic parsers
* Refactor some methods to prove the system
* Beginnings of generated JS support

### CODE REVIEW

File: src/app/Api/LocalReport.php [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<?php namespace TmlpStats\Api;
+
+// This is an API servicer. All API methods are simple static methods
+// that take typed input and return array responses.
+use TmlpStats\Http\Controllers;
+use TmlpStats\Reports\Arrangements;
+use TmlpStats\StatsReport;
+use App;
+
+class LocalReport {
```
Comment (pdarg) [inline src/app/Api/LocalReport.php]: Coding standards comment: please put opening braces for classes and class methods on the next line

File: src/resources/views/reports/tools/api_controller_output.blade.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class ApiController extends ApiControllerBase {
+    protected $methods = [
+@foreach ($apiMethodsFlat as $method)
+        "{{ $method->absName }}" => "{{ $method->absNameLocal() }}",
+@endforeach
+    ];
+
+@foreach ($apiMethodsFlat as $method)
+<? $callName = str_replace('.', '->', $method->absName)  ?>
+    function {{ $method->absNameLocal() }}($input) {
```
Comment (pdarg) [inline src/resources/views/reports/tools/api_controller_output.blade.php]: will you make these public

Comment (crast) [inline src/resources/views/reports/tools/api_controller_output.blade.php]: nope!

**Comment by crast** [comment]
Comment for slack!

**Pull request merged**